### PR TITLE
Features/cnx 1107 migrate to channex pci

### PR DESCRIPTION
**Pull request by EevanW**

### CODE REVIEW

File: lib/ex_wubook/pci_proxies/channex_pci.ex [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+      code = response.status
+
+      meta =
+        %{}
+        |> add_cards_info(response.headers)
+        |> finalize_meta(%{response: response.body, status_code: code})
+
+      cond do
+        code == 200 -> {:ok, response.body, meta}
+        code == 404 -> {:error, :not_found, Map.update(meta, :errors, [], &[:not_found | &1])}
```
Comment (DmitryKK) [inline lib/ex_wubook/pci_proxies/channex_pci.ex:150]: Why do you write errors in meta? We can see error code in two element of tuple

File: lib/ex_wubook/pci_proxies/channex_pci.ex [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    with {:ok, response} <- request(:post, url, body, [], []) do
+      code = response.status
+
+      meta =
+        %{}
+        |> add_cards_info(response.headers)
+        |> finalize_meta(%{response: response.body, status_code: code})
+
+      cond do
+        code == 200 -> {:ok, response.body, meta}
```
Comment (DmitryKK) [inline lib/ex_wubook/pci_proxies/channex_pci.ex:149]: what about code > 200 ?

Comment (EevanW) [reply lib/ex_wubook/pci_proxies/channex_pci.ex:149]: It's copy -paste from pci_booking.ex now it's work, I don't want change it.

File: lib/ex_wubook/pci_proxies/channex_pci.ex (hunk 1 above)
Comment (EevanW) [reply lib/ex_wubook/pci_proxies/channex_pci.ex:150]: It's copy-paste for backward compatibility with PCIBooking provider module.

File: lib/ex_wubook/pci_proxies/channex_pci.ex [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+  defp get_url(url, api_key) do
+    "#{pci_url()}?apikey=#{api_key}&method=post&profile=wubook&url=#{url}"
+  end
+
+  defp get_card_token(token, lcode, booking_code, password, api_key) do
+    url = get_card_token_url(api_key)
+    body = get_card_token_body(token, lcode, booking_code, password)
+
+    with {:ok, body, headers} <- send_request(url, body),
```
Comment (anronin) [inline lib/ex_wubook/pci_proxies/channex_pci.ex]:
```suggestion
    with {:ok, body, meta} <- send_request(url, body),
```

File: lib/ex_wubook/pci_proxies/channex_pci.ex [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+         {:ok, %{param: [0, card_data]}} <- XMLRPC.decode(body) do
+      {
+        :ok,
+        %{
+          cardholder_name: card_data["cc_owner"],
+          card_number: card_data["cc_number"],
+          expiration_date: NormalizeExpirationDate.execute(card_data["cc_expiring"]),
+          card_type: @card_types[card_data["cc_type"]],
+          cvv: card_data["cc_cvv"],
+          token: headers[:token]
```
Comment (anronin) [inline lib/ex_wubook/pci_proxies/channex_pci.ex]:
```suggestion
          token: meta[:token]
```

File: lib/ex_wubook/pci_proxies/channex_pci.ex [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+        <param><value><int>#{lcode}</int></value></param>
+        <param><value><int>#{booking_code}</int></value></param>
+        <param><value><string>#{password}</string></value></param>
+      </params>
+    </methodCall>
+    """
+  end
+
+  defp send_request(url, body) do
+    with {:ok, response} <- request(:post, url, body, [], []) do
```
Comment (anronin) [inline lib/ex_wubook/pci_proxies/channex_pci.ex]:
```suggestion
    with {:ok, response} <- post(url, body, [], []) do
```

File: lib/ex_wubook/pci_proxies/channex_pci.ex (hunk 3 above)
Comment (EevanW) [reply lib/ex_wubook/pci_proxies/channex_pci.ex]: Changed

File: lib/ex_wubook/pci_proxies/channex_pci.ex (hunk 4 above)
Comment (EevanW) [reply lib/ex_wubook/pci_proxies/channex_pci.ex]: Changed

File: lib/ex_wubook/pci_proxies/channex_pci.ex (hunk 5 above)
Comment (EevanW) [reply lib/ex_wubook/pci_proxies/channex_pci.ex]: Changed

File: lib/ex_wubook/pci_proxies/channex_pci.ex (hunk 2 above)
Comment (Judis) [reply lib/ex_wubook/pci_proxies/channex_pci.ex:149]: By spec, they not use any other success code except 200

**Review by Judis — APPROVED** [review]

**Pull request merged**